TiFlash crash for DynamicThreadPool::scheduleTask

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
```
[2022/11/25 04:34:41.447 +08:00] [ERROR] [BaseDaemon.cpp:377] [########################################] [source=BaseDaemon] [thread_id=5417]
[2022/11/25 04:34:41.447 +08:00] [ERROR] [BaseDaemon.cpp:378] ["(from thread 1861) Received signal Segmentation fault(11)."] [source=BaseDaemon] [thread_id=5417]
[2022/11/25 04:34:41.447 +08:00] [ERROR] [BaseDaemon.cpp:408] ["Address: 0x38"] [source=BaseDaemon] [thread_id=5417]
[2022/11/25 04:34:41.447 +08:00] [ERROR] [BaseDaemon.cpp:414] ["Access: read."] [source=BaseDaemon] [thread_id=5417]
[2022/11/25 04:34:41.447 +08:00] [ERROR] [BaseDaemon.cpp:423] ["Address not mapped to object."] [source=BaseDaemon] [thread_id=5417]
[2022/11/25 04:34:41.480 +08:00] [ERROR] [BaseDaemon.cpp:570] ["\n       0x554cd21\tfaultSignalHandler(int, siginfo_t*, void*) [tiflash+89443617]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:221\n  0x7f780532f630\t<unknown symbol> [libpthread.so.0+63024]\n       0x16f2d82\tbool boost::lockfree::queue<DB::MPMCQueue<std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >*>::pop<DB::MPMCQueue<std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >*>(DB::MPMCQueue<std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >*&) [tiflash+24063362]\n                \t/usr/local/bin/../include/c++/v1/atomic:0\n       0x16f1b7a\tDB::DynamicThreadPool::scheduleTask(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >) [tiflash+24058746]\n                \tdbms/src/Common/DynamicThreadPool.cpp:63\n       0x16ee400\tauto DB::DynamicThreadPool::schedule<std::__1::function<void ()> >(bool, std::__1::function<void ()>&&) [tiflash+24044544]\n                \tdbms/src/Common/DynamicThreadPool.h:58\n       0x16ee007\tnon-virtual thunk to DB::(anonymous namespace)::DynamicThreadManager::schedule(bool, std::__1::function<void ()>) [tiflash+24043527]\n                \tdbms/src/Common/ThreadManager.cpp:0\n       0x591c039\tDB::MergingAndConvertingBlockInputStream::readImpl() [tiflash+93438009]\n                \tdbms/src/Interpreters/Aggregator.cpp:1875\n       0x5577295\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+89617045]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x5576f85\tDB::IProfilingBlockInputStream::read() [tiflash+89616261]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x582f039\tDB::ParallelAggregatingBlockInputStream::readImpl() [tiflash+92467257]\n                \tdbms/src/DataStreams/ParallelAggregatingBlockInputStream.cpp:125\n       0x5577295\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+89617045]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x5576f85\tDB::IProfilingBlockInputStream::read() [tiflash+89616261]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x171a481\tDB::SharedQueryBlockInputStream::fetchBlocks() [tiflash+24224897]\n                \tdbms/src/DataStreams/SharedQueryBlockInputStream.h:153\n       0x16eed2b\tDB::ExecutableTask<std::__1::packaged_task<void ()> >::execute() [tiflash+24046891]\n                \tdbms/src/Common/ExecutableTask.h:52\n       0x16f2273\tDB::DynamicThreadPool::executeTask(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&) [tiflash+24060531]\n                \tdbms/src/Common/DynamicThreadPool.cpp:101\n       0x16f1f46\tDB::DynamicThreadPool::dynamicWork(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >) [tiflash+24059718]\n                \tdbms/src/Common/DynamicThreadPool.cpp:142\n       0x16f34f2\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, std::__1::thread DB::ThreadFactory::newThread<void (DB::DynamicThreadPool::*)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, void (DB::DynamicThreadPool::*&&)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*&&, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&&)::'lambda'(auto&&...), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > > >(void*) [tiflash+24065266]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f7805327ea5\tstart_thread [libpthread.so.0+32421]"] [source=BaseDaemon] [thread_id=5417]
[2022/11/25 04:34:41.480 +08:00] [ERROR] [BaseDaemon.cpp:377] [########################################] [source=BaseDaemon] [thread_id=5417]
[2022/11/25 04:34:41.480 +08:00] [ERROR] [BaseDaemon.cpp:378] ["(from thread 1421) Received signal Segmentation fault(11)."] [source=BaseDaemon] [thread_id=5417]
[2022/11/25 04:34:41.480 +08:00] [ERROR] [BaseDaemon.cpp:408] ["Address: 0x38"] [source=BaseDaemon] [thread_id=5417]
[2022/11/25 04:34:41.480 +08:00] [ERROR] [BaseDaemon.cpp:414] ["Access: read."] [source=BaseDaemon] [thread_id=5417]
[2022/11/25 04:34:41.480 +08:00] [ERROR] [BaseDaemon.cpp:423] ["Address not mapped to object."] [source=BaseDaemon] [thread_id=5417]
[2022/11/25 04:34:41.480 +08:00] [ERROR] [BaseDaemon.cpp:570] ["\n       0x554cd21\tfaultSignalHandler(int, siginfo_t*, void*) [tiflash+89443617]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:221\n  0x7f780532f630\t<unknown symbol> [libpthread.so.0+63024]\n       0x16f2d82\tbool boost::lockfree::queue<DB::MPMCQueue<std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >*>::pop<DB::MPMCQueue<std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >*>(DB::MPMCQueue<std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >*&) [tiflash+24063362]\n                \t/usr/local/bin/../include/c++/v1/atomic:0\n       0x16f1b7a\tDB::DynamicThreadPool::scheduleTask(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >) [tiflash+24058746]\n                \tdbms/src/Common/DynamicThreadPool.cpp:63\n       0x16ee400\tauto DB::DynamicThreadPool::schedule<std::__1::function<void ()> >(bool, std::__1::function<void ()>&&) [tiflash+24044544]\n                \tdbms/src/Common/DynamicThreadPool.h:58\n       0x16ee007\tnon-virtual thunk to DB::(anonymous namespace)::DynamicThreadManager::schedule(bool, std::__1::function<void ()>) [tiflash+24043527]\n                \tdbms/src/Common/ThreadManager.cpp:0\n       0x591bd34\tDB::MergingAndConvertingBlockInputStream::readImpl() [tiflash+93437236]\n                \tdbms/src/Interpreters/Aggregator.cpp:1891\n       0x5577295\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+89617045]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x5576f85\tDB::IProfilingBlockInputStream::read() [tiflash+89616261]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x582f039\tDB::ParallelAggregatingBlockInputStream::readImpl() [tiflash+92467257]\n                \tdbms/src/DataStreams/ParallelAggregatingBlockInputStream.cpp:125\n       0x5577295\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+89617045]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x5576f85\tDB::IProfilingBlockInputStream::read() [tiflash+89616261]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x171a481\tDB::SharedQueryBlockInputStream::fetchBlocks() [tiflash+24224897]\n                \tdbms/src/DataStreams/SharedQueryBlockInputStream.h:153\n       0x16eed2b\tDB::ExecutableTask<std::__1::packaged_task<void ()> >::execute() [tiflash+24046891]\n                \tdbms/src/Common/ExecutableTask.h:52\n       0x16f2273\tDB::DynamicThreadPool::executeTask(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&) [tiflash+24060531]\n                \tdbms/src/Common/DynamicThreadPool.cpp:101\n       0x16f1f46\tDB::DynamicThreadPool::dynamicWork(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >) [tiflash+24059718]\n                \tdbms/src/Common/DynamicThreadPool.cpp:142\n       0x16f34f2\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, std::__1::thread DB::ThreadFactory::newThread<void (DB::DynamicThreadPool::*)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, void (DB::DynamicThreadPool::*&&)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*&&, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&&)::'lambda'(auto&&...), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > > >(void*) [tiflash+24065266]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f7805327ea5\tstart_thread [libpthread.so.0+32421]"] [source=BaseDaemon] [thread_id=5417]
```
### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiFlash version? (Required)
master e7be8e7f06af59a0359287b9849bdd436666e3b4

--- a/dbms/src/Server/Server.cpp
+++ b/dbms/src/Server/Server.cpp
@@ -1284,8 +1284,6 @@ int Server::main(const std::vector<std::string> & /*args*/)
 
             main_config_reloader.reset();
             users_config_reloader.reset();
-
-            DynamicThreadPool::global_instance.reset();
         });
 
         /// This object will periodically calculate some metrics.

